src/common/system/proc_parser.h

```diff
@@ -330,6 +330,34 @@ class ProcParser {
    */
   StatusOr<absl::flat_hash_set<std::string>> GetMapPaths(pid_t pid) const;
 
+  /**
+   * ProcessMaps tracks details from /proc/<pid>/maps
+   */
+  struct ProcessMap {
+      uint64_t vmem_start = 0;
+      uint64_t vmem_end = 0;
+      // TODO(ddelnano): Change this to an octal like number like chmod
+      /* uint8_t permissions = 0; */
+      std::string permissions = "";
```

Comment:
I've removed the `= "";` and this TODO since I wasn't sure if changing the `GetMapEntries` code would be more performant if it was backed by an 8 bit field.